src/components/Item.jsx

```diff
@@ -51,12 +45,53 @@ const Item = ({
     return estimatedInterval;
   };
 
+  const checkDate = (lastPurchasedDate) => {
+    if (lastPurchasedDate === null) {
+      return false;
+    } else {
+      const day = 60 * 60 * 24;
+      const now = Date.now() / 1000;
+      return now - lastPurchasedDate.seconds < day;
+    }
+  };
+
   const checked = checkDate(lastPurchasedDate);
-  const className = checked ? 'checked' : '';
+  const checkedClassName = checked ? 'checked' : '';
+
+  const groupItemClassName = () => {
+    if (
+      numberOfPurchases < 2 ||
+      (lastPurchasedDate != null &&
+        calculateLatestInterval() >= 2 * nextPurchase)
+    ) {
+      return 'inactive';
+    }
+    if (nextPurchase < 7) {
+      return 'soon';
+    } else if (nextPurchase >= 7 && nextPurchase <= 30) {
+      return 'kind-of-soon';
+    } else if (nextPurchase > 30) {
+      return 'not-so-soon';
+    }
```

Comment:
I was looking at that too, but it looks like js requires constant cases.
```

Reply:
@kevinfalting Great observation. Yes, usually `switch` statement is used for matching expressions  (the `switch`) against different values (the `case`).

However, there is a `switch(true)` construction method that allows us to match not only against values but against expressions as well. Expressions will be evaluated before matching, so any expression that evaluates to `true` will be matched. I have recently learned about this too.  Here is a [great read-up about `switch(true)` construction](https://seanbarry.dev/posts/switch-true-pattern).

This is how it would work in our case: 

```
  const groupItemClassName = () => {
    if (
      lastPurchasedDate != null &&
      calculateLatestInterval() >= 2 * nextPurchase
    ) {
      return 'inactive';
    }
    switch (true) {
      case nextPurchase < 7:
        return 'soon';
      case nextPurchase >= 7 && nextPurchase <= 30:
        return 'kind-of-soon';
      default:
        return 'not-so-soon';
    }
```

Reply:
Well hott dam, look at that. The MDN doc linked made no mention of it. 😝